packages/thrift-client/src/main/connections/HttpConnection.ts

```diff
@@ -220,47 +206,50 @@ export class HttpConnection extends Core.ThriftConnection<RequestOptions> {
                 : this.url
 
         // Merge user options with required options
-        const requestOptions: RequestOptions & UrlOptions = Core.overlayObjects(
+        const requestOptions: OptionsOfBufferResponseBody = Core.overlayObjects(
             this.requestOptions,
             options,
             {
                 method: 'POST',
                 body: dataToWrite,
-                encoding: null, // Needs to be explicitly set to null to get Buffer in response body
                 url: requestUrl,
                 headers: {
-                    'Content-Length': dataToWrite.length,
+                    'Content-Length': dataToWrite.length.toString(),
                     'Content-Type': 'application/octet-stream',
                 },
-            },
+            } as RequestOptions,
```

Comment:
is casting to RequestOptions unavoidable here?

Reply:
normally using `as` is less type safe, but in this case it was used to make sure the right properties were set